src/attribute.cpp

```diff
@@ -147,6 +148,9 @@ Encrypted::Encrypted(uint8_t type, const uint8_t* data, size_t size, const std::
             mdBuffer[j + secret.length()] = data[i * 16 + j];
     }
 
+    const auto deleteObject = std::remove(plaintext.begin(), plaintext.end(), 0);
+    plaintext.erase(deleteObject, plaintext.end());
```

Comment:
No need in `deleteObject` variable. Use it inline:
```c++
plaintext.erase(std::remove(plaintext.begin(), plaintext.end(), 0), plaintext.end());
```

This is classic C++ 'Remove-Erase' idiom.

Reply:
On the other hand, I think current behavior is still wrong. Consider a string like this: `"test\0abc"`. Your algorithm will produce `"testabc"`, when the correct should be just `"test"`.

Consider finding a first zero element in the range and use it instead of `plaintext.end()` in the `assign` below.